### PR TITLE
NIAD-1199: Map Deviation Indicator

### PR DESCRIPTION
**Pull request by saileshp-kainos**
## Description

Created a mapping for the `Deviation Indicator` (`HI`, `LO`, `PA`, `OR`) values from the LaboratoryInvestigationResult segment (`RSL+NV+111.1++:::fL+HI'`) in the `Observation` resource in the `interpretation` property.

Have updated the UAT test data to reflect this and created a unit test to test the interpretation mapping.

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1199

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [X] Acceptance Criteria met
- [X] Commit messages are meaningful
- [X] Manually tested
- [ ] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [ ] If your pull request depends on any other, please link them in the description
- [ ] main branch is passing/green on CI
- [ ] Add/update any relevant documentation

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java [hunk 1]
```diff
@@ -178,9 +179,20 @@ private void mapStatus(final LabResult labResult, final Observation observation)
...
+                    observation.getInterpretation().setText(
+                        deviation.getCode() + " : " + deviation.getDescription())
+                )
+            );
+        }
+
         private void mapLaboratoryInvestigationResult(final LabResult labResult, final Observation observation) {
             // Observation.value.valueQuantity.*
             labResult.getInvestigationResult().ifPresent(investigationResult -> {
+
```
Comment (saileshp-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java:195]: Remove useless whitespace

File: src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java [hunk 2]
```diff
@@ -178,6 +179,16 @@ private void mapStatus(final LabResult labResult, final Observation observation)
...
         }
 
+        private void mapInterpretation(final LabResult labResult, final Observation observation) {
+            // Observation.interpretation = SG18.RSL.7857
+            labResult.getInvestigationResult().ifPresent(investigationResult ->
+                Optional.ofNullable(investigationResult.getDeviatingResultIndicator()).ifPresent(deviation ->
+                    observation.getInterpretation().setText(
+                        deviation.getCode() + " : " + deviation.getDescription())
+                )
+            );
```
Comment (andrewwa-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java:189]:
```suggestion
            labResult.getInvestigationResult()
                .map(LaboratoryInvestigationResult::getDeviatingResultIndicator)
                .map(deviation -> deviation.getCode() + " : " + deviation.getDescription())
                .ifPresent(deviation -> observation.getInterpretation().setText(deviation));
```
`Optional#map` can safely go from something present to being empty.
```
labResult.getInvestigationResult() // Optional<LaboratoryInvestigationResult>
    .map(LaboratoryInvestigationResult::getDeviatingResultIndicator) // Optional<DeviatingResultIndicator>
    .map(deviation -> deviation.getCode() + " : " + deviation.getDescription()) // Optional<String>
    .ifPresent(deviation -> observation.getInterpretation().setText(deviation));
```

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java [hunk 3]
```diff
@@ -193,6 +194,30 @@ void testMapLaboratoryInvestigationCodedResult() {
...
+        final var observations = mapper.mapToObservations(message, mockPatient, emptyMap(), mockOrganization,
+            mockPractitioner);
+
+        assertThat(observations).hasSize(1)
+            .first()
+            .extracting(Observation::getInterpretation)
+            .extracting(CodeableConcept::getText)
+            .satisfies(result ->
+                assertThat(result).isEqualTo(DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT.getCode()
+                    + " : " + DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT.getDescription()));
```
Comment (andrewwa-kainos) [inline src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java:218]: Can use `.isEqualTo` directly on an AbstractObjectAssert - no need to go through `.satisfies`. Also, I'd prefer just to see the raw string being put in the FHIR output.

```suggestion
        assertThat(observations).hasSize(1)
            .first()
            .extracting(Observation::getInterpretation)
            .extracting(CodeableConcept::getText)
            .isEqualTo("HI : Above high reference limit");
```

**Review by andrewwa-kainos — CHANGES_REQUESTED** [review]

**Review by andrewwa-kainos — APPROVED** [review]

**Review by antonlebedjko — APPROVED** [review]
LGTM

**Pull request merged**